### ISSUE TITLE
uncaught exceptions are not getting captured in the console logs

### ISSUE THREAD

**Issue by agnivade** [chromedp/chromedp#374]
#### What versions are you running?

<pre>
$ go list -m github.com/chromedp/chromedp
github.com/chromedp/chromedp v0.3.0
$ google-chrome --version
Google Chrome 72.0.3626.109
$ go version
go version go1.12.4 linux/amd64
</pre>

#### What did you do? Include clear steps.

- I have an html file which tries to call an undefined function.
- I am serving the file through a local web server.

#### What did you expect to see?
When I load the file using chromedp, I do not see the error captured in the code.

#### What did you see instead?
No log in the output

But if I just add a console.error("something") above the line, that shows up fine.

```html
<!doctype html>
<html>

<head>
	<meta charset="utf-8">
	<title>Go wasm</title>
</head>

<body>
	<script>
			// console.error("got msg"); If you uncomment this line, "got msg" clearly shows up.
			hello();
	</script>
</body>
</html>
```

The Go code:

```go
package main

import (
	"context"
	"fmt"
	"log"
	"os"
	"strconv"
	"time"

	cdpruntime "github.com/chromedp/cdproto/runtime"
	"github.com/chromedp/chromedp"
)

func main() {
	logger := log.New(os.Stderr, "", log.LstdFlags|log.Lshortfile)

	// create chrome instance
	timeout, cancel := context.WithTimeout(context.Background(), 5*time.Second)
	defer cancel()
	ctx, cancel := chromedp.NewContext(
		timeout,
	)
	defer cancel()

	chromedp.ListenTarget(ctx, func(ev interface{}) {
		switch ev := ev.(type) {
		case *cdpruntime.EventConsoleAPICalled:
			for _, arg := range ev.Args {
				s, err := strconv.Unquote(string(arg.Value))
				if err != nil {
					logger.Println(err)
					continue
				}
				fmt.Printf("%s\n", s)
			}
		}
	})

	err := chromedp.Run(ctx,
		chromedp.Navigate(`http://localhost:8080`),
	)
	if err != nil {
		logger.Fatal(err)
	}
}
```

Basically, any uncaught exception does not show up. This came up when I was using this library to make [wasmbrowsertest](https://github.com/agnivade/wasmbrowsertest). But CORS issues were not getting logged. I had to manually run the code in the browser to see that it was a CORS issue.

Is there any technical limitation due to which this is happening ?

From the documentation - I see https://chromedevtools.github.io/devtools-protocol/tot/Runtime#event-exceptionThrown to be there which should give you the details. But I don't see anything like that in the code.

**Comment by mvdan** [comment]
Thanks for following the issue template - nice for a change :)

Your `ListenTarget` only handles console calls, such as `console.log`. If you also made it handle `Runtime.exceptionThrown`, you could make that do whatever you want. So that's working as intended, as far as I can see.

On the other hand, you could argue that chromedp should log exceptions by default, since they would help users spot bugs quicker. That isn't happening right now. Is this what you mean by this issue?

**Comment by agnivade** [comment]
Ah my bad for not reading the docs well enough. The reason I thought `cdpruntime.EventConsoleAPICalled` would capture it because `console.error`s were getting captured with it, and not just `console.log` . So I thought something must be missing for exceptions, because the browser logs it as a console error.

> On the other hand, you could argue that chromedp should log exceptions by default, since they would help users spot bugs quicker. 

I think you could add exception handling here in this example - https://godoc.org/github.com/chromedp/chromedp#example-ListenTarget--ConsoleLog

That should handle the most common cases and would make it clear to the user that there is a separate event for exception.

Thanks for the help :)

**Comment by yehudamakarov** [comment]
What about mixed content warnings in the chrome browser? How can I pick those up in a similar fashion?
